packages/backend/src/routes/CommentRouter.ts

```diff
@@ -246,3 +254,142 @@ async function createCommentSubsidy(req, res) {
         comment,
     })
 }
+
+async function editComment(req, res) {
+    const id = req.params.id
+    const unirepSocialContract = new ethers.Contract(
+        UNIREP_SOCIAL,
+        UNIREP_SOCIAL_ABI,
+        DEFAULT_ETH_PROVIDER
+    )
+
+    // Parse Inputs
+    const { publicSignals, proof, content } = req.body
+    const epkProof = new EpochKeyProof(
+        publicSignals,
+        formatProofForSnarkjsVerification(proof)
+    )
+    const newHashedContent = ethers.utils.keccak256(
+        ethers.utils.toUtf8Bytes(content)
+    )
+
+    const error = await verifyEpochKeyProof(req.db, epkProof)
+    if (error !== undefined) {
+        res.status(422).json({
+            error,
+        })
+        return
+    }
+
+    const { hashedContent: oldHashedContent, commentId } = await req.db.findOne(
+        'Comment',
+        {
+            where: {
+                _id: id,
+            },
+        }
+    )
+
+    const calldata = unirepSocialContract.interface.encodeFunctionData('edit', [
+        commentId,
+        oldHashedContent,
+        newHashedContent,
+        epkProof.publicSignals,
+        epkProof.proof,
+    ])
+
+    const hash = await TransactionManager.queueTransaction(
+        unirepSocialContract.address,
+        {
+            data: calldata,
+        }
+    )
+
+    await req.db.update('Comment', {
+        where: {
+            _id: id,
+            commentId,
+            hashedContent: oldHashedContent,
+        },
+        update: {
+            content,
+            hashedContent: newHashedContent,
+        },
+    })
+
+    const comment = await req.db.findOne('Comment', { where: { _id: id } })
+
+    res.json({
+        error: error,
+        transaction: hash,
+        comment,
+    })
+}
+
+async function deleteComment(req, res) {
```

Comment:
We're not actually deleting the comment here, which makes sense. Instead of having a custom backend route why don't we have the frontend update the comment content to `Deleted by user.` or similar. This delete function is a bit misleading